addons/website/static/src/js/editor/wysiwyg.js

```diff
@@ -86,6 +86,28 @@ const WebsiteWysiwyg = Wysiwyg.extend({
             }
         }
 
+        // Restore device-based "Invisible Elements".
+        let isMobile;
+        this.trigger_up('service_context_get', {
+            callback: (ctx) => {
+                isMobile = ctx['isMobile'];
+            },
+        });
+        for (const el of this.$editable[0].querySelectorAll('.d-none')) {
+            if ([...el.classList].some(className => className.startsWith('d-md-'))) {
+                el.classList.add('o_snippet_mobile_invisible');
+                if (isMobile) {
+                    el.dataset.invisible = '1';
+                }
+            }
+        }
+        for (const el of this.$editable[0].querySelectorAll('.d-md-none')) {
+            el.classList.add('o_snippet_desktop_invisible');
+            if (!isMobile) {
+                el.dataset.invisible = '1';
+            }
+        }
```

Comment:
The code for the invisible element is normally confined to the snippets.editor.js file, I don't think that should change. If I am not mistaken:
- Adding `o_snippet_XXX_invisible` classes on editor start should not be needed, those should be saved with the DOM?
- Setting theright `invisible` attribute is normally already automatic for all `.o_snippet_XXX_invisible` element?

We may need a migration script though, indeed.

Reply:
Indeed, I misunderstood: I somehow was convinced that those classes and the `invisible` attribute were not saved.